### PR TITLE
Add inline example for grdview

### PR DESCRIPTION
**Pull request by willschlitzer**
Addresses #1957 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/grdview.py [hunk 1]
```diff
@@ -117,6 +119,39 @@ def grdview(self, grid, **kwargs):
...
+    ...     region=[-92.5, -82.5, -3, 7],
+    ...     registration="gridline",
+    ... )
+    >>> # create a new plot with pygmt.Figure()
+    >>> fig = pygmt.Figure()
+    >>> # create the contour plot
+    >>> fig.grdview(
+    ...     # pass in the grid downloaded above
+    ...     grid=grid,
+    ...     # set the perspective to have an azimuth of 130 and elevation of 30
```
Comment (michaelgrund) [inline pygmt/src/grdview.py:139]:
```suggestion
    ...     # set the perspective to have an azimuth of 130° and an elevation of 30°
```

**Comment by seisman** [comment]
Ping @willschlitzer to see if you have time to finish this PR before the v0.9.0 release.

**Comment by seisman** [comment]
It's likely that Will is busy recently, so I'm taking over this PR to make sure that we can have it done before v0.9.0.

File: pygmt/src/grdview.py [hunk 2]
```diff
@@ -117,6 +119,39 @@ def grdview(self, grid, **kwargs):
...
+    ...     perspective=[130, 30],
+    ...     # add a frame to the x- and y-axes
+    ...     # specify grid labels on the south and east borders of the plot
+    ...     frame=["xa", "ya", "wSnE"],
+    ...     # set the projection to Mercator for the 10 cm figure
+    ...     projection="M10c",
+    ...     # set the vertical scale (z-axis) to 2 cm
+    ...     zsize="2c",
+    ...     # set surface plot to plot CPT
+    ...     surftype="s",
```
[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/src/grdview.py]: I've rephrased it to

> set "surface plot" to color the surface via a CPT

File: pygmt/src/grdview.py [hunk 3]
```diff
@@ -117,6 +119,39 @@ def grdview(self, grid, **kwargs):
...
+    Example
+    -------
+    >>> import pygmt
+    >>> # load the 15 arc-minutes grid with "gridline" registration
+    >>> # in a specified region
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m",
+    ...     region=[-92.5, -82.5, -3, 7],
+    ...     registration="gridline",
+    ... )
```
Comment (yvonnefroehlich) [inline pygmt/src/grdview.py:132]:
```suggestion
    >>> # load the 30 arc-minutes grid with "gridline" registration
    >>> # in a specified region
    >>> grid = pygmt.datasets.load_earth_relief(
    ...     resolution="30m",
    ...     region=[-92.5, -82.5, -3, 7],
    ...     registration="gridline",
    ... )
```

File: pygmt/src/grdview.py [hunk 4]
```diff
@@ -117,6 +119,39 @@ def grdview(self, grid, **kwargs):
...
+    >>> # create a new plot with pygmt.Figure()
+    >>> fig = pygmt.Figure()
```
Comment (yvonnefroehlich) [inline pygmt/src/grdview.py:134]:
```suggestion
    >>> # create a new figure instance with pygmt.Figure()
    >>> fig = pygmt.Figure()
```

File: pygmt/src/grdview.py [hunk 5]
```diff
@@ -117,6 +119,39 @@ def grdview(self, grid, **kwargs):
...
+    >>> fig.grdview(
+    ...     # pass in the grid downloaded above
+    ...     grid=grid,
+    ...     # set the perspective to an azimuth of 130° and an elevation of 30°
+    ...     perspective=[130, 30],
+    ...     # add a frame to the x- and y-axes
+    ...     # specify annotations on the south and east borders of the plot
+    ...     frame=["xa", "ya", "wSnE"],
+    ...     # set the projection to Mercator for the 10 cm figure
+    ...     projection="M10c",
```
Comment (yvonnefroehlich) [inline pygmt/src/grdview.py:145]:
```suggestion
    ...     # set the projection to Mercator for the 10 cm wide figure
    ...     projection="M10c",
```

File: pygmt/src/grdview.py [hunk 6]
```diff
@@ -117,6 +119,39 @@ def grdview(self, grid, **kwargs):
...
+    >>> fig.grdview(
+    ...     # pass in the grid downloaded above
+    ...     grid=grid,
+    ...     # set the perspective to an azimuth of 130° and an elevation of 30°
+    ...     perspective=[130, 30],
+    ...     # add a frame to the x- and y-axes
+    ...     # specify annotations on the south and east borders of the plot
+    ...     frame=["xa", "ya", "wSnE"],
+    ...     # set the projection to Mercator for the 10 cm wide figure
+    ...     projection="M10c",
```
Comment (yvonnefroehlich) [inline pygmt/src/grdview.py:145]: Hm. Unforantly I feel my review in https://github.com/GenericMappingTools/pygmt/pull/2381#discussion_r1141282033 was not precise 🙁. This is a 3-D figure and width and hight of the figure change when changing the `perspective` argument. Probably this width refers to the map or plot itself? 
```suggestion
    ...     # set the projection to Mercator for the 10 cm wide plot
    ...     projection="M10c",
```

Comment (seisman) [reply pygmt/src/grdview.py:145]: `M10c` means the frame width is 10-cm in 2D view. Thus, in 3D view, the actually width can be any other values.

**Comment by willschlitzer** [comment]
Hey, sorry that I've been MIA; I should have time to work on things this week, but no problem with @seisman making changes/approving things.

File: pygmt/src/grdview.py (hunk 6 above)
Comment (yvonnefroehlich) [reply pygmt/src/grdview.py:145]: Ah, OK. This makes sense.

File: pygmt/src/grdview.py [hunk 7]
```diff
@@ -117,6 +119,39 @@ def grdview(self, grid, **kwargs):
...
+    >>> # create the contour plot
+    >>> fig.grdview(
+    ...     # pass in the grid downloaded above
+    ...     grid=grid,
+    ...     # set the perspective to an azimuth of 130° and an elevation of 30°
+    ...     perspective=[130, 30],
+    ...     # add a frame to the x- and y-axes
+    ...     # specify annotations on the south and east borders of the plot
+    ...     frame=["xa", "ya", "wSnE"],
+    ...     # set the projection to Mercator for the 10 cm wide figure
```
Comment (seisman) [inline pygmt/src/grdview.py:144]:
```suggestion
    ...     # set the projection of the 2D map to Mercator with a 10 cm width
```

File: pygmt/src/grdview.py (hunk 6 above)
Comment (seisman) [reply pygmt/src/grdview.py:145]: See my comment https://github.com/GenericMappingTools/pygmt/pull/2381/files#r1141549661.

**Comment by seisman** [comment]
> Hey, sorry that I've been MIA; I should have time to work on things this week, but no problem with @seisman making changes/approving things.

Don't worry. Welcome back.

File: pygmt/src/grdview.py [hunk 8]
```diff
@@ -117,6 +119,39 @@ def grdview(self, grid, **kwargs):
...
+    >>> # create the contour plot
+    >>> fig.grdview(
+    ...     # pass in the grid downloaded above
+    ...     grid=grid,
+    ...     # set the perspective to an azimuth of 130° and an elevation of 30°
+    ...     perspective=[130, 30],
+    ...     # add a frame to the x- and y-axes
+    ...     # specify annotations on the south and east borders of the plot
+    ...     frame=["xa", "ya", "wSnE"],
+    ...     # set the projection of the 2D map to Mercator with a 10 cm width
```
Comment (yvonnefroehlich) [inline pygmt/src/grdview.py:144]:
```suggestion
    ...     # set the projection of the 2-D map to Mercator with a 10 cm width
```

**Review by seisman — APPROVED** [review]

**Review by yvonnefroehlich — APPROVED** [review]

**Review by yvonnefroehlich — COMMENTED** [review]
Hm. Just wondering whether we should decide about starting comments with either upper-case letters or with lower-case letters, across all examples? I have a little preference towards upper-case letters, but I am also totally fine with the lower-case option 🙂.

**Comment by seisman** [comment]
> Hm. Just wondering whether we should decide about starting comments with either upper-case letters or with lower-case letters, across all examples? I have a little preference towards upper-case letters, but I am also totally fine with the lower-case option slightly_smiling_face.

Either is fine to me.

**Pull request merged**